### DISCUSSION TITLE
Plagiatrism is flourishing?

### DISCUSSION THREAD

**Discussion by NickVasenkov** [codewars/codewars.com#3050] [General]
It's my first day here, and I only did 12 Kata up to 4kyu, but I think I see a pattern:

The most upvoted solution has the most similar (VERY similar) solutions grouped with it.

Therefore, I can conclude that a common strategy to get Honor is just by copying solutions of others.

How bad does this compromise the ratings? Is it still possible to achieve a high Honor percentile, knowing that I can't solve problems as fast as other people can copy them?

Are there better resources to demonstrate one's programming and Data Science skills with a similar idea (small challenges of different difficulty that increase one's rank)?

**Comment by ansis-m** [comment]
Sometimes there is just one or few obvious, elegant and coinscise solutions. On many ocassions my solutions get grouped with other top 3 solutions even though I have never cheated. Acctually, I dont think that cheating is that prevalent.

**Reply by fayod7** [reply]
7

**Comment by hobovsky** [comment]
> The most upvoted solution has the most similar (VERY similar) solutions grouped with it.
> 
> Therefore, I can conclude that a common strategy to get Honor is just by copying solutions of others.

This is a very limited and quite short-sighted conclusion. There is many legit ways to end up with a grouped solution:
- Code golf kata may have a very small variety of possible solutions.
- For some simple tasks it's really possible to end up with identical solutions. If a solution is just a call to a few built-ins, it's possible that many users come up with calling the said built-ins.
- Someone can submit a solution of another person after they solve the kata. It happens quite often that someone solves a kata on their own, checks other solutions, sees some interesting one, and checks how it works, and submits it. They dont break any rules, because they already solved the kata before copying the other solution. This happens all the time, and it's highly probable that highly upvoted solution will catch an eye of many users who will want to test it out.

So no, it does not automatically mean that a user who got their solution grouped with another, older solution, cheated or was dishonest in any way. It _may_, but it's not necessarily so.

**Comment by RobertJClose** [comment]
I haven't been using CodeWars for that long, but to me the purpose of the site is to help people make a habit of writing great code, not to prove who is the best coder via an Honor rating. 

Most definitions of "kata" involve improvement through repetition of some perfect form, so to me if you're not copying the best solutions (after solving it yourself), then you're not doing it right!

**Comment by Chrono79** [comment]
I'd say 12 katas is too small as a sample to assert that.

**Comment by RTSRLLC** [comment]
Copy pasting defeats the purpose of why we came here: to write code that solves a problem and become better coders. 
A non cheater will grow their skills.

**Comment by juanRCoder** [comment]
It depends on each one, if you copy or not, in the future you will see your results

**Comment by NickVasenkov** [comment]
Thanks for the advice!
________________________________

**Comment by aryuemaan** [comment]
It's true that on platforms like Codewars (where "Kata" are used), similar solutions often get grouped, and it can seem like copying is a path to Honor.
